### ISSUE TITLE
[Custom Public Directoiry] icongenie and other extension i.e @quasar/qpdfviewer generating file to public folder

### ISSUE THREAD

**Issue by dipaksarkar** [quasarframework/quasar#14677]
### What happened?

I'm using vite and changed the publicDir by extendVitConf() 

```
...
extendViteConf(viteConf) {
        Object.assign(viteConf, {
          publicDir: path.resolve(__dirname, "./statics"),
        });
},
...
```

When I have run command to generate icons and installing extensions it's  not generating file to my custom public directory.

### What did you expect to happen?

It should be generate the icons and related files of extension to custom public directory.

### Reproduction URL

https://github.com/dipaksarkar/qaravel

### How to reproduce?

# Qaravel (qaravel)

Qaravel is a Combination of Laravel and Quasar

## 1. Add FontAwesome pro key

We used Fontawesome pro version.

## 2. Install the dependencies

```bash
yarn
```

## 3. Add Configuration

```bash
cp .env.example .env
cp .env.frontend .env.frontend.dev
```

## 4. Configuration for Laravel Sanctum

We have used Laravel Sanctum SPA authentication. Laravel Sanctum provides a featherweight
authentication system for SPAs (single page applications), mobile applications, and simple, token
based APIs. Sanctum allows each user of your application to generate multiple API tokens for their
account. These tokens may be granted abilities / scopes which specify which actions the tokens are
allowed to perform. Please check laravel default
[sanctum configuration](https://laravel.com/docs/9.x/sanctum#spa-authentication)

```bash
SANCTUM_STATEFUL_DOMAINS=qaravel.gomedia:9000 (for local you have to include your port)
SESSION_DOMAIN=.qaravel.gomedia
```

## 5. Migrate the Database

Add your database details on .env

```
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=qaravel
DB_USERNAME=root
DB_PASSWORD=toor
```

```bash
yarn fresh:dev
```

### Steps for Windows User

Press Win key + R > Paste following in the box > Press Enter

```
c:\windows\system32\drivers\etc\hosts
```

Then add thoes line at the bottom and save as admin.

```
127.0.0.1   api.qaravel.gomedia
127.0.0.1   app.qaravel.gomedia
```

### 7. Start the app in development mode (hot-code reloading, error reporting, etc.)

> Only for windows user

```
php artisan serve --host=api.qaravel.gomedia  --port=80
```

Change your api url from .env.frontend.dev

```
API_URL=http://api.qaravel.gomedia
```

```bash
yarn start:web
```

### Build the app for production

```bash
yarn build:web
```

### Lint the files

```bash
yarn run lint
```

### How to deply it in local?

Valet is a Laravel development environment for Mac minimalists. No Vagrant, no /etc/hosts file. You
can even share your sites publicly using local tunnels. Yeah, we like it too. We used valet to
deploy this project. Check the configuration of [Laravel Valet](https://laravel.com/docs/9.x/valet)

### How to deply it in web?

First build your app then move all files in server and run following command

```bash
cp .htaccess.example .htaccess
```

### Customize the configuration

See [Configuring quasar.conf.js](https://quasar.dev/quasar-cli/quasar-conf-js).

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite), Plugins (quasar), Icon Genie CLI (@quasar/icongenie)

**Comment by rstoenescu** [comment]
Thank you for the report, and apologies for the long silence on it.

After looking into this, we are going to close it as not planned. Here is the reasoning:

`extendViteConf()` is a hook that runs inside `quasar dev` / `quasar build`, after Quasar has fully resolved its own configuration. It hands you the final Vite config to tweak. It is not a Quasar project setting, so nothing outside that in-process build step can see what you change there:

- Icon Genie is a standalone CLI package. It only looks for `quasar.config.*` to find your project root; it never loads the file, and it has no way to run the Vite hooks. It writes to `public/` because that is the Quasar CLI convention for static assets.
- App Extensions are installed by `quasar ext`, which also does not go through the Vite pipeline. Where an extension writes its files is decided by the extension author, using the standard project layout.
- Even within `@quasar/app-vite`, the public folder is pinned to `<app>/public` and used independently of Vite's `publicDir` (PWA manifest resolution, the BEX public-folder watcher, `build.ignorePublicFolder`). Overriding `publicDir` in `extendViteConf()` only changes what Vite copies; the rest of the tooling still assumes `public/`, so the two can silently disagree.

In short, a custom public folder is not a supported project layout, and we don't plan to add one: it would have to be threaded through the CLI, Icon Genie, every App Extension and the docs for a very small gain over the default.

Recommended approach: keep `public/` as the folder Quasar tooling writes to. If your Laravel setup needs the static files elsewhere, either point Laravel at `public/` or copy them over in a post-build step (a small script in your `package.json`), which keeps every Quasar tool working unchanged.

Thanks again for taking the time to report this.
